File: Leetcode/Medium/53.go
// https://leetcode.com/problems/maximum-subarray/?envType=study-plan-v2&envId=top-interview-150

func maxSubArray(nums []int) int {
    curmax := 0
    ans := -100000
    for _, v := range nums {
        curmax += v
        if curmax > ans {
            ans = curmax
        }
        if curmax < 0 {
            curmax = 0
        }
    }

    return ans
}

File: Leetcode/Medium/53.java
// https://leetcode.com/problems/maximum-subarray/

class Solution {
    public int maxSubArray(int[] nums) {
        int answer = nums[0], current = 0;
        for (int x : nums) {
            current = Math.max(current + x, x);
            answer = Math.max(answer, current);
        }
        return answer;
    }
}